packages/gatsby/cache-dir/navigation.js

```diff
@@ -18,9 +18,8 @@ function maybeRedirect(pathname) {
 
   if (redirect != null) {
     if (process.env.NODE_ENV !== `production`) {
-      const pageResources = loader.loadPageSync(pathname)
-
-      if (pageResources != null) {
```

Comment:
Seems to me like these cases are not identical.

`isPageNotFound` will return `this.pageDb.get(pagePath) && this.pageDb.get(pagePath).notFound === true`
`loadPageSync` will return `this.pageDb.has(pagePath) ? this.pageDb.get(pagePath).payload : undefined`

The `isPageNotFound` is, it seems, referring to an existing page that represents a `404` page (or whatever). Whereas the check here is trying to assert whether we have anything to show for the requested path at all (whether it be content or a 404 page or anything else).

It's not clear to me that it was your intention to change that semantic.

Reply:
The thing is here: `loadPageSync` never returned undefined whether you have this file inside pages directory or not. So that means condition always was true and it showed error message.

I ll go with example: lets say I want to have redirect  from `path-1` to `path-2`, and I only have `path-2.js` file under pages folder. It redirected as expected but with error message saying: The route path-2 matches a page and a redirect but I don't have `path-1.js` file under pages. Somehow 
 `loadPageSync('path-1')` returned object which is true for if condition on line 22.